musical_chairs/routines.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import random
+import collections
+import math
+
+from arms import ArmBernoulli
+from .users import SecondaryUser
+
+
+def mc_routine(n_users, params, n_arms, t_horizon, arm_means):
+    ''' Apply muscial chairs algorithm to a pb with t_horizon time steps.
+        Args:
+            - n_users (int): number of users.
+            - n_arms (int): number of arms.
+            - params (dict): t0 - exploring phase / t1 - exploiting phase
+            - t_horizon (int): time steps.
+            - plot (Bool): True if plot is needed
+        Output:
+            - total_rewards (ndarray): total reward at each time step.
+    '''
+    arms = list()
+    for i in range(n_arms):
+        arms.append(ArmBernoulli(arm_means[i]))
+    users = [SecondaryUser(n_arms, n_users, t_horizon) for i in range(n_users)]
+    total_rewards = np.zeros((t_horizon, 1))
+    if t_horizon < params['t1']:
+        return print("horizon must be at least t1")
```

Comment:
oula ça marche ça ?
Tu devrais plutôt raise une erreur : http://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python